File: exercism.io/javascript/grains/grains.js
//
// This is only a SKELETON file for the 'Grains' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

import bigInt from './lib/big-integer';

export class Grains {
  square(num) {
    let result = bigInt(2).pow(num-1).toString()
    return result
  }

  total() {
    let sum = "0"
    for(let i = 0; i < 64; i++){
      sum = bigInt(bigInt(2).pow(i)).add(sum).toString()
    }
    return sum
  }
}

File: exercism.io/csharp/grains/Grains.cs
using System;

public static class Grains
{
    public static ulong Square(int n)
    {
        if (n < 1 || n > 64) { throw new ArgumentOutOfRangeException(); }
        return (ulong) Math.Pow(2.00, (double)n - 1);
    }

    public static ulong Total()
    {
        ulong sum = 0UL;
        ulong accumulate = 1;

        for(int i = 0; i < 64; i++)
        {
            sum += accumulate;
            accumulate *= 2;
        }
        return sum;
        
    }
}